File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

map<char, int> table;

vector<int> solution(string s) {
    vector<int> answer;
    
    for (int i = 0; i < s.size(); i++) {
        char cur = s[i];
        if (table.find(cur) == table.end()) {
            table[cur] = i;
            answer.push_back(-1);
        } else {
            answer.push_back(i - table[cur]);
            table[cur] = i;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    n = len(s)
    di = {}
    for i in range(n):
        cur = s[i]
        if (cur not in di):
            di[cur] = i
            answer.append(-1);
        else:
            answer.append(i - di[cur])
            di[cur] = i
    
    return answer